text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,671 @@
+requirements for host tools. This review and approval may take place in a PR
+adding the target's host tools to CI, by an infrastructure team member
+reporting the outcome of a team discussion, or by including the infrastructure
+team in the RFC proposing the target.
+
+- Tier 1 targets with host tools should typically include all of the additional
+  tools such as `clippy` and `rustfmt`, unless there is a target-specific
+  reason why a tool cannot possibly make sense for the target.
+  - Unlike with tier 2, for tier 1 we will not exclude specific tools on the
+    sole basis of them being less likely to be used; rather, we'll take that
+    into account when considering whether the target should be at tier 1 with
+    host tools. In general, on any tier 1 target with host tools, people
+    should be able to expect to find and install all the same components that
+    they would for any other tier 1 target with host tools.
+- Approval of host tools will take into account the additional time required to
+  build the host tools, and the substantial additional storage required for the
+  host tools.
+- Host tools for the target must have substantial, widespread interest within
+  the developer community, and must serve the ongoing needs of multiple
+  production users of Rust across multiple organizations or projects. These
+  requirements are subjective, and determined by consensus of the approving
+  teams. This requirement will be evaluated independently from the
+  corresponding tier 1 requirement; it is possible for a target to have
+  sufficient interest for cross-compilation, but not have sufficient interest
+  for native compilation. The host tools may be dropped if they no longer meet
+  this requirement, even if the target otherwise qualifies as tier 1.
+- The host tools must build, run, and pass tests reliably in CI.
+  - The target must not disable an excessive number of tests or pieces of tests
+    in the testsuite in order to do so. This is a subjective requirement.
+- Building the host tools and running the testsuite for the host tools must not
+  take substantially longer than other targets.
+  - In particular, if building the target's host tools takes a reasonable
+    amount of time, but the target cannot run the testsuite in a timely fashion
+    due to low performance of either native code or accurate emulation, that
+    alone may prevent the target from qualifying as tier 1 with host tools.
+- New tier 1 targets with host tools must not have a hard requirement for
+  signed, verified, or otherwise "approved" binaries. Developers must be able
+  to build, run, and test binaries for the target on systems they control.
+  (Doing so may require enabling some appropriate "developer mode" on such
+  systems, but must not require the payment of any additional fee or other
+  consideration, or agreement to any onerous legal agreements.)
+  - While the Rust project may supply appropriately signed tools, this
```

Comment:
Hm, I have a minor nit here: while tier 2 requires availability of easily accessible signing methods in our CI, this statement here ("may supply") is rather weak. It becomes unclear to me what our policy to signing for platforms that require signing for smooth operation (particularly macOS, Windows) is. I would have expected tier 1 to make a stronger statement towards users.

Reply:
@skade I've clarified the language in this requirement, and I've moved it to apply to all tier 1 targets, not just to tier 1 with host tools. Thanks!